listener/wal_transaction.go

```diff
@@ -217,7 +217,11 @@ func (w *WalTransaction) CreateActionData(
 			valueType: rel.Columns[num].valueType,
 			isKey:     rel.Columns[num].isKey,
 		}
-		column.AssertValue(row.Value)
+		if row.IsUnchangedToastedValue && len(oldRows) > num {
+			column.AssertValue(oldRows[num].Value)
+		} else {
+			column.AssertValue(row.Value)
+		}
```

Comment:
The `len(oldRows) > num` guards against tables that don't have `ALTER TABLE REPLICA IDENTITY FULL`. Those tables won't have any oldRows (dataOld) and will cause an index out of bounds exception.

Reply:
We discussed this IRL and figured we'd bring it to the PR for discussion: if we don't have full replica identity, but `row.IsUnchangedToastedValue`, should we perhaps 🔥 the key entirely from the pubsub? The end result will be the same in Elasticsearch right now, but I think the pubsub message is at least more true to reality.